src/control/lib/control/pool.go

```diff
@@ -612,7 +612,37 @@ func PoolQuery(ctx context.Context, rpcClient UnaryInvoker, req *PoolQueryReq) (
 	}
 
 	pqr := new(PoolQueryResp)
-	return pqr, convertMSResponse(ur, pqr)
+	err = convertMSResponse(ur, pqr)
+	if err != nil {
+		return nil, err
+	}
+
+	err = pqr.UpdateState()
+	if err != nil {
+		return nil, err
+	}
+
+	return pqr, err
+}
+
+// Update the pool state
+func (pqr *PoolQueryResp) UpdateState() error {
```

Comment:
Is there a ticket to address this on the engine side by adding a degraded state at the engine level? It would be good to add a TODO referencing that ticket number here. Whoever works on that code will need to know this can be removed.